emoji/core.py

```diff
@@ -86,9 +85,22 @@ def replace(match):
 
     # Decode shortcuts
     string = get_emoji_regexp(no_space).sub(replace, string)
+    
+    if max_length:
+        safe_length = _get_safe_length(string, max_length)
+        string = string[0:safe_length]
 
     return string
 
+def _get_safe_length(text, max_length):
+        previous = 0
+        for m in get_emoji_shortcode_regex().finditer(text):
+            if m.start() + len(m.group()) > max_length:
+                return m.start() if "emoji_modifier_fitzpatrick" not in m.group() else previous
```

Comment:
The fitzpatrick modifiers are special characters that can be appended to emojis to change their skin tones. I added this check to avoid separating an emoji from its skin tone, thereby converting it to the default yellow version.